### PR TITLE
docs: update air-gapped docs

### PR DESCRIPTION
**Pull request by itaysk**
discussed here https://github.com/aquasecurity/trivy/issues/7029#issuecomment-2225913815

I did my best to restructure the docs in a way that makes sense, I have a few open comments:

1. not sure how authentication works with `--checks-bundle-repository` @simar7 
2. do we have a "manually populate cache" option for checks? @simar7 
3. inconsistency in naming between `--checks-bundle-repository` and `--skip-check-update`, not sure we want to fix this at this point.
4. `--offline-scan` isn't clearly explained, I presume this is about maven central? then the name is misleading (should be java specific). again, not sure we want to fix now but wanted to bring to attention. Also I thought this can be expanded to disable all the external dependencies (all the DBs) in one flag.

### CODE REVIEW

File: docs/docs/advanced/air-gap.md [hunk 1]
```diff
@@ -1,142 +1,125 @@
...
-    trivy --cache-dir $TRIVY_TEMP_DIR image --download-java-db-only
-    tar -cf ./javadb.tar.gz -C $TRIVY_TEMP_DIR/java-db metadata.json trivy-java.db
-    rm -rf $TRIVY_TEMP_DIR
-    ```
-=== "oras >= v0.13.0"
-    Please follow [oras installation instruction][oras].
+For Trivy DB, configure it in the [same way as for private images](../advanced/private-registries/index.md).
 
-    Download `javadb.tar.gz`:
+For Java DB, you need to run `docker login YOUR_REGISTRY`. Currently, specifying a username and password is not supported.
```
Comment (knqyf263) [inline docs/docs/advanced/air-gap.md:54]: The original document was incorrect. The vulnerability DB and Java DB work the same, and it supports only `docker login` now.

File: docs/docs/advanced/air-gap.md [hunk 2]
```diff
@@ -1,142 +1,125 @@
...
 
-=== "oras >= v0.13.0"
-    Please follow [oras installation instruction][oras].
+## Running Trivy in air-gapped environment
+In an air-gapped environment, you have to tell Trivy on every scan to not attempt to download the latest database files, otherwise the scan will fail. The following flags are relevant:
 
-    Download `db.tar.gz`:
+- `--skip-db-update` to skip updating the main vulnerability database.
+- `--skip-java-db-update` to skip updating the Java vulnerability database.
+- `--offline-scan` to scan Java applications without issuing API requests.
```
Comment (knqyf263) [inline docs/docs/advanced/air-gap.md:24]: We no longer need it and can delete it for now. But @DmitriyLewen suggested re-implementing sonatype API. We'll add it back after that.

Comment (knqyf263) [reply docs/docs/advanced/air-gap.md:24]: Also, users suggested that `--offline` enables all needed flags for air-gapped environments.

**Review by knqyf263 — COMMENTED** [review]
>--offline-scan isn't clearly explained, I presume this is about maven central? then the name is misleading (should be java specific). again, not sure we want to fix now but wanted to bring to attention. Also I thought this can be expanded to disable all the external dependencies (all the DBs) in one flag.

`--offline-scan` should affect all scanning, not limited to JAR scanning (To complicate things, JAR scanning was switched to Java DB, so this flag is not used for anything now.). When we implement internet access in the future, it should be disabled by `--offline-scan`. Also, I think misconfiguration scanning should respect this flag (downloading Terraform modules, etc.).

**Comment by nikpivkin** [comment]
1. The checks bundle is distributed as an OCI artifacts and retrieved from the Artifact registry. Authentication methods are described [here](https://aquasecurity.github.io/trivy/v0.53/docs/advanced/private-registries/#credential).

File: docs/docs/advanced/air-gap.md [hunk 3]
```diff
@@ -1,142 +1,125 @@
...
+trivy image \
+    --db-repository myregistry.local/trivy-db \
+    --java-db-repository myregistry.local/trivy-java-db \
+    --offline-scan \
+    --checks-bundle-repository myregistry.local/trivy-checks \
+    myimage
+```
 
-=== "Trivy"
+### Authentication
```
[the remark this answers is not in the document]
Comment (itaysk) [reply docs/docs/advanced/air-gap.md:50]: I agree, I also thought about the allowlist and private access not belonging to the airgap document, but also didn't want to leave them in a vulnerability specific document since they are shared to misconfig and maybe other things in the future.
WDYT about renaming the air-gap document to "advanced database scenarios"? then it makes more sense why the network requirements, authentication, and offline settings are there. 
It does sounds similar to the existing "configuration/db.md" doc that you linked, but I thought that the entire "configuration" section is purely CLI flags reference (and I thought it's auto-generated)

**Comment by itaysk** [comment]
> The checks bundle is distributed as an OCI artifacts and retrieved from the Artifact registry. Authentication methods are described [here](https://aquasecurity.github.io/trivy/v0.53/docs/advanced/private-registries/#credential).

@nikpivkin , @knqyf263 just [said](https://github.com/aquasecurity/trivy/pull/7160#discussion_r1677319315) that the vuln DBs support only docker login, they don't support user/pass. is it the same for checks bundle as well? or checks bundle does support user/pass unlike vuln scanner?

**Comment by nikpivkin** [comment]
@itaysk supports login and password

File: docs/docs/advanced/air-gap.md (hunk 1 above)
Comment (nikpivkin) [reply docs/docs/advanced/air-gap.md:54]: @knqyf263 Why does Java DB only support authentication via docker login?  When an artifact is loaded, [this](https://github.com/aquasecurity/trivy/blob/main/pkg/oci/artifact.go#L73) method is called, which loads the image with authentication.

File: docs/docs/advanced/air-gap.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/docs/advanced/air-gap.md:50]: >WDYT about renaming the air-gap document to "advanced database scenarios"? then it makes more sense why the network requirements, authentication, and offline settings are there.

Yes, It makes more sense, but since we also have checks bundle, advanced "database" scenarios might be misleading.

File: docs/docs/advanced/air-gap.md (hunk 1 above)
Comment (knqyf263) [reply docs/docs/advanced/air-gap.md:54]: It's my bad. I was using an old laptop today and referring to an old commit.

Comment (itaysk) [reply docs/docs/advanced/air-gap.md:54]: so just to confirm, vuln db, java db, checks db all support the same authentication, and that is trivy's "private registry" options. right?

File: docs/docs/advanced/air-gap.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (itaysk) [reply docs/docs/advanced/air-gap.md:58]: https://github.com/aquasecurity/trivy/pull/7160/commits/4695eebf7a876ff18c9a56f07e4ef4c141dc7e10

File: docs/docs/advanced/air-gap.md (hunk 1 above)
Comment (itaysk) [reply docs/docs/advanced/air-gap.md]: https://github.com/aquasecurity/trivy/pull/7160/commits/4695eebf7a876ff18c9a56f07e4ef4c141dc7e10

File: docs/docs/advanced/air-gap.md (hunk 2 above)
Comment (itaysk) [reply docs/docs/advanced/air-gap.md]: https://github.com/aquasecurity/trivy/pull/7160/commits/4695eebf7a876ff18c9a56f07e4ef4c141dc7e10

File: docs/docs/advanced/air-gap.md [hunk 4]
```diff
@@ -1,142 +1,128 @@
...
 ```
 
-[allowlist]: ../references/troubleshooting.md
-[oras]: https://oras.land/docs/installation
+### Java DB
+
+For Java DB the process is the same, except for the following:
+1. Image location is `ghcr.io/aquasecurity/trivy-java-db:1`
+2. Archive file name is `javadb.tar.gz`
+3. DB file name is `trivy-java.db`
```
Comment (simar7) [inline docs/docs/advanced/air-gap.md:124]: Formatting needs a fix

image

File: docs/docs/scanner/misconfiguration/check/builtin.md [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
-Built-in checks are mainly written in [Rego][rego] and Go.
-Those checks are managed under [trivy-checks repository][trivy-checks].
+## Checks Sources
+Trivy has an extensive library of misconfiguration checks that is maintained at <https://github.com/aquasecurity/trivy-checks>.  
+Trivy checks are mainly written in [Rego][rego], while some checks are written in Go.  
 See [here](../../../coverage/iac/index.md) for the list of supported config types.
 
-For suggestions or issues regarding policy content, please open an issue under the [trivy-checks][trivy-checks] repository.
+## Checks Bundle
+When performing a misconfiguration scan, Trivy will automatically downloads the relevant Checks bundle. The bundle is cached locally and Trivy will reuse it for subsequent scans on the same machine. Trivy takes care of updating the cache automatically so normally can be oblivious to it.
```
Comment (simar7) [inline docs/docs/scanner/misconfiguration/check/builtin.md:9]:
```suggestion
When performing a misconfiguration scan, Trivy will automatically download the relevant Checks bundle. The bundle is cached locally and Trivy will reuse it for subsequent scans on the same machine. Trivy takes care of updating the cache automatically, so the user can normally can be oblivious to it.
```

File: docs/docs/advanced/air-gap.md (hunk 1 above)
Comment (simar7) [reply docs/docs/advanced/air-gap.md]: Yes

**Comment by itaysk** [comment]
@knqyf263 wdyt about removing the manual cache manipulation guide? it seems very internal and i think using the flag is better

**Comment by knqyf263** [comment]
@itaysk We discussed it in https://github.com/aquasecurity/trivy/issues/7029 and agreed not to add flags.

**Comment by knqyf263** [comment]
Or do you mean we should force users to upload DB to their OCI registry?

**Comment by itaysk** [comment]
I meant the latter. not saying it's a big problem, just seems redundant and makeshift to me. ok i'll keep it, but then I want to confirm what is the procedure for misconfig checks (@simar7 @nikpivkin )?

**Comment by itaysk** [comment]
Added another node about loading config from directory using `--config-check` flag. Also found many outdates references to this flag which I fixed in the same commit. FTR, also adding this to the list of inconsistencies (checks vs check) @simar7 can you please review

File: docs/docs/scanner/misconfiguration/check/builtin.md [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
-For suggestions or issues regarding policy content, please open an issue under the [trivy-checks][trivy-checks] repository.
+## Checks Bundle
+When performing a misconfiguration scan, Trivy will automatically download the relevant Checks bundle. The bundle is cached locally and Trivy will reuse it for subsequent scans on the same machine. Trivy takes care of updating the cache automatically, so normally users can be oblivious to it.
 
-## Check Distribution
-Trivy checks are distributed as an OPA bundle on [GitHub Container Registry][ghcr] (GHCR).
-When misconfiguration detection is enabled, Trivy pulls the OPA bundle from GHCR as an OCI artifact and stores it in the cache.
-Those checks are then loaded into Trivy OPA engine and used for detecting misconfigurations.
-If Trivy is unable to pull down newer checks, it will use the embedded set of checks as a fallback. This is also the case in air-gap environments where `--skip-policy-update` might be passed.
+For CLI flags related to the database, please refer to [this page](../configuration/db.md).
```
Comment (nikpivkin) [inline docs/docs/scanner/misconfiguration/check/builtin.md:11]: Database?

File: docs/docs/scanner/misconfiguration/index.md [hunk 7]
```diff
@@ -326,13 +326,13 @@ trivy config --misconfig-scanners=terraform,dockerfile .
 
 Will only scan for misconfigurations that pertain to Terraform and Dockerfiles.
 
-### Passing custom checks
-You can pass policy files or directories including your custom checks through `--policy` option.
+### Loading custom checks
+You can load policy files or directories including your custom checks through `--config-check` flag.
```
Comment (simar7) [inline docs/docs/scanner/misconfiguration/index.md]: For some reason, GitHub has seemed to disable suggestions for me. But in this line it should be `load check files` instead of policy.

File: docs/docs/scanner/misconfiguration/custom/data.md [hunk 8]
```diff
@@ -31,5 +31,5 @@ Then, you need to pass data paths through `--data` option.
 Trivy recursively searches the specified paths for JSON (`*.json`) and YAML (`*.yaml`) files.
 
 ```bash
-$ trivy conf --policy ./policy --data data --namespaces user ./configs
+$ trivy conf --config-check ./checks --data ./data --namespaces user ./configs
```
Comment (simar7) [inline docs/docs/scanner/misconfiguration/custom/data.md:34]: It is a nit but I like to use `trivy config` instead of `trivy conf` as it is more descriptive.

File: docs/docs/scanner/misconfiguration/custom/index.md [hunk 9]
```diff
@@ -2,10 +2,10 @@
 
 ## Overview
 You can write custom checks in [Rego][rego].
-Once you finish writing custom checks, you can pass the policy files or the directory where those policies are stored with `--policy` option.
+Once you finish writing custom checks, you can pass the policy files or the directory where those checks are stored with --config-check` option.
```
Comment (simar7) [inline docs/docs/scanner/misconfiguration/custom/index.md:5]:
```suggestion
Once you finish writing custom checks, you can pass the check files or the directory where those checks are stored with --config-check` option.
```

**Comment by simar7** [comment]
> I meant the latter. not saying it's a big problem, just seems redundant and makeshift to me. ok i'll keep it, but then I want to confirm what is the procedure for misconfig checks (@simar7 @nikpivkin )?

The process would be the same as trivy-db, the users have to navigate Trivy's cache to place the checks in the right place in case of an air gap solution.

Maybe we should document this cache structure in more detail as a whole as it is a common place for all scanners?

**Comment by itaysk** [comment]
@knqyf263 I've addressed vex hub in air gapped env, let me know if it's ok

File: docs/docs/advanced/air-gap.md [hunk 10]
```diff
@@ -1,142 +1,162 @@
...
+
+For Java DB the process is the same, except for the following:
+
+1. Image location is `ghcr.io/aquasecurity/trivy-java-db:1`
+2. Archive file name is `javadb.tar.gz`
+3. DB file name is `trivy-java.db`
+
+## Misconfigurations scanning
+
+Note that the misconfigurations database is also embedded in the Trivy binary (at build time), and will be used as a fallback if the external database is not available. This means that you can still scan for misconfigurations in an air-gapped environment using the Checks from the time of the Trivy release you are using.
```
Comment (knqyf263) [inline docs/docs/advanced/air-gap.md:160]: Do we call the checks bundle "database"?

**Review by knqyf263 — COMMENTED** [review]
LGTM. Left some comments.

File: docs/docs/advanced/air-gap.md (hunk 10 above)
Comment (itaysk) [reply docs/docs/advanced/air-gap.md:160]: I have some thoughts about this, but don't want to delay this PR with them so I'll do change it

File: docs/docs/advanced/air-gap.md
Comment (knqyf263) [inline docs/docs/advanced/air-gap.md:1]: I have an idea about re-structuring this page, but we can do that in another PR.

**Review by knqyf263 — APPROVED** [review]

**Review by simar7 — APPROVED** [review]
